### ISSUE TITLE
Suggestion: move include/* into include/oos/* ?

### ISSUE THREAD

**Issue by saki7** [zussel/oos#63]
Reasons for the suggestion:
- Many of the existing libraries follow the rule that the include directory should be located under "include/**library_name**"
  - So that `include_directories("path_to_library/include")` yields inclusion path `"library_name/*"` properly
- The current directory structure breaks up certain IDE's auto-completion :fire: 

The change requires the all `#include` statement to be fixed:

```cpp
#include "object/has_many_vector.hpp"
```

(↑ No offense, but it seems like a pretty bad idea to me. The path may be resolved to some library called "**object**" since the resolving process for the `#include` statement is implementation-defined)

to

```cpp
#include "oos/object/has_many_vector.hpp"
```

---

And also move `db/` to `include/oos/db/` and change the `#include` statement from

```cpp
#include "mysql_connection.hpp"
```

to

```cpp
#include "oos/db/mysql/mysql_connection.hpp"
```

I know it will take quite a effort to do this, but I'd strongly suggest it to be changed.
I think also the CMake install script needs modification for this change.

**Comment by zussel** [comment]
I'm thinking about targeting this issue for a 0.5.1 release instead of 0.5.0.

What do you think?

**Comment by saki7** [comment]
I think this major fix would not be a "patchlevel" change - it breaks all remaining PRs and feature candidates since the files will actually be moved to different locations. So IMO this fix should be done in 0.5.0. The sooner the better.

**Comment by zussel** [comment]
Thanks for your opinion! Indeed it would break significantly the project layout and therefor all future release. I'll do it in 0.5.0.

As far  as I can see it is already installed under ```include/oos``` but the backends must be adjusted and all includes in the sources. Maybe I can fiddle a cool ```sed``` statement for this job :smile:
